### PR TITLE
Datasearch: support searching for moves by target and use sendReplyBox for movesearchhelp

### PR DESCRIPTION
**Pull request by AnnikaCodes**
This was suggested [here](https://www.smogon.com/forums/threads/add-targets-as-a-parameter-on-ms.3665018/).

### CODE REVIEW

File: server/chat-plugins/datasearch.ts [hunk 1]
```diff
@@ -1026,6 +1031,23 @@ function runMovesearch(target: string, cmd: string, canAll: boolean, message: st
...
+		'userandallies': 'allies',
+		'usersside': 'allySide',
+		'usersteam': 'allyTeam',
+		'any': 'any',
+		'opponentsside': 'foeSide',
+		'oneadjacent': 'normal',
+		'randomadjacent': 'randomNormal',
+		'scripted': 'scripted',
+		'user': 'self',
+	};
```
Comment (KrisXV) [inline server/chat-plugins/datasearch.ts:1050]:
```suggestion
	const allTargets: {[k: string]: string} = {
		oneally: 'adjacentAlly',
		userorally: 'adjacentAllyOrSelf',
		oneadjacentopponent: 'adjacentFoe',
		all: 'all',
		alladjacent: 'allAdjacent',
		alladjacentopponents: 'allAdjacentFoes',
		userandallies: 'allies',
		usersside: 'allySide',
		usersteam: 'allyTeam',
		any: 'any',
		opponentsside: 'foeSide',
		oneadjacent: 'normal',
		randomadjacent: 'randomNormal',
		scripted: 'scripted',
		user: 'self',
	};
```

**Comment by Zarel** [comment]
Does this depend on #6756?

File: server/chat-plugins/datasearch.ts [hunk 2]
```diff
@@ -1089,6 +1111,26 @@ function runMovesearch(target: string, cmd: string, canAll: boolean, message: st
 				continue;
 			}
 
+			if (target.substr(0, 'targets '.length) === 'targets ') {
```
Comment (Zarel) [inline server/chat-plugins/datasearch.ts:1114]:
```suggestion
			if (target.startsWith('targets ')) {
```

**Comment by AnnikaCodes** [comment]
> Does this depend on #6756?

It does not.

**Pull request merged**